extensions/test-extension/src/custom-context-menu/findingsContextMenu.ts

```diff
@@ -0,0 +1,100 @@
+const findingsContextMenu = {
+  id: 'cornerstoneContextMenu',
+  customizationType: 'ohif.contextMenu',
+  menus: [
+    {
+      id: 'forExistingMeasurement',
+      // selector restricts context menu to when there is nearbyToolData
+      selector: ({ nearbyToolData }) => !!nearbyToolData,
+      items: [
+        {
+          customizationType: 'ohif.contextSubMenu',
+          label: 'Site',
+          actionType: 'SubMenu',
+          subMenu: 'siteSelectionSubMenu',
+        },
+        {
+          customizationType: 'ohif.contextSubMenu',
+          label: 'Finding',
+          actionType: 'SubMenu',
+          subMenu: 'findingSelectionSubMenu',
+        },
+        {
+          // customizationType is implicit here in the configuration setup
+          label: 'Delete Measurement',
+          commands: [
+            {
+              commandName: 'deleteMeasurement',
+            },
+          ],
+        },
+        {
+          label: 'Add Label',
+          commands: [
+            {
+              commandName: 'setMeasurementLabel',
+            },
+          ],
+        },
+
+        // The example below shows how to include a delegating sub-menu,
+        // Only available on the @ohif/hp-extension.mn hanging protocol
+        // To demonstrate, select the 3x1 layout from the protocol menu
+        // and right click on a measurement.
+        {
+          label: 'IncludeSubMenu',
+          selector: ({ protocol }) => protocol?.id === '@ohif/hp-extension.mn',
+          delegating: true,
+          subMenu: 'orientationSelectionSubMenu',
```

Comment:
How is this `subMenu` when actionType is not `SubMenu`?

Reply:
Because subMenu is the name of another menu, in this case the delegating subMenu, and this item delegates to the sub-menu, it never gets run by itself.